CMakeLists.txt

```diff
@@ -11,14 +11,22 @@ include_directories(
     ${PROJECT_SOURCE_DIR}/src
 )
 
-add_library(${PROJECT_NAME} SHARED
+add_library(chessboard_lib STATIC
     src/Chessboard.cpp
     src/Chessboard.h
     src/ChessboardCorner.h
     src/ChessboardQuad.h
     src/Spline.h
+    src/chessBoardCornersDetector.cpp
 )
+set_target_properties(chessboard_lib PROPERTIES POSITION_INDEPENDENT_CODE ON)
 
-add_executable(test_chessboard src/test_chessboard.cpp)
-target_link_libraries(test_chessboard ${PROJECT_NAME})
-target_link_libraries(test_chessboard ${OpenCV_LIBS})
+add_subdirectory(pybind11)
+pybind11_add_module(pyChessBoardDetector src/chessBoardCornersDetector_pybind.cpp)
+
+add_executable(chessBoardCornersDetector src/chessBoardCornersDetector.cpp)
```

Comment:
this will fail. 
There is no main function in this file. so you cannot create an executable for this file

Please test before sending for review.